File: src/satellite/src/bookkeeping.rs
use crate::types::BookKeepingData;
use crate::types::BookKeepingStatus::TransferFromDone;
use ic_cdk::id;
use junobuild_satellite::{set_doc_store, SetDoc};
use junobuild_utils::encode_doc_data;

pub fn save_icp_transferred(key: String) -> Result<(), String> {
    let book_data: BookKeepingData = BookKeepingData {
        status: TransferFromDone,
    };

    let data = encode_doc_data(&book_data)?;

    let doc: SetDoc = SetDoc {
        data,
        description: None,
        version: None,
    };

    let _ = set_doc_store(id(), "bookkeeping".to_string(), key, doc)?;

    Ok(())
}

File: src/satellite/bookkeeping.ts
import type { BookKeepingData } from '$lib/types/datastore';
import type { SetDoc } from '@junobuild/functions';
import { id } from '@junobuild/functions/ic-cdk';
import { encodeDocData, setDocStore } from '@junobuild/functions/sdk';

export const saveIcpTransferred = async (key: string) => {
	const bookData: BookKeepingData = {
		status: 'transfer-from-done'
	};

	const data = encodeDocData(bookData);

	const doc: SetDoc = {
		data
	};

	setDocStore({
		caller: id(),
		key,
		collection: 'bookkeeping',
		doc
	});
};
